Add complete call sequence (#176)
sharetribe/flex-docs: src/docs/background/16-how-saving-payment-card-works-in-ftw/index.md

component. That component allows you to also select one-time payment
instead (and optionally replace the current default payment method with
this new payment card). If there's no default payment method saved,
one-time payment form is shown on its own. One-time payment needs card
details, card holder's name etc. and optional permission to save card
details for future bookings.

### Submit StripePaymentForm

After submitting `StripePaymentForm`, there are up to 5 calls in
sequence (to Flex and Stripe APIs):

#### Step 1.

```js
=> sdk.transactions.initiate({ processAlias, transition: 'transition/request-payment', ...})
```

What happens behind the scene:

- API creates PaymentIntent against Stripe API and returns
  stripePaymentIntentClientSecret inside transaction's protectedData.
- Booking is created at this step - so, availability management will
  block dates for conflicting bookings.
- Automatic expiration happens in 15 minutes, if process is not
  transitioned to 'transition/confirm-payment' before that.
- After this call, created transaction is saved to session storage in
  FTW (or existing enquiry tx is updated).

> **When you save card details**, a new parameter needs to be passed if

Reviewer: ```suggestion
**When you intend to save card details**, a new parameter needs to be passed if
```

card details are saved at the same time: `setupPaymentMethodForSaving`

```js
=> sdk.transactions.initiate({ processAlias, transition: 'transition/request-payment', params: { listingId, bookingStart, bookingEnd, setupPaymentMethodForSaving: true }})
```

> **When you use previously saved payment card**, the id of Stripe's

Reviewer: ```suggestion
**When you are using previously saved payment card**, the id of Stripe's
```

payment method needs to be sent to Flex API as `paymentMethod`, when
requesting payment.

```js

> => sdk.transactions.initiate({ processAlias, transition: 'transition/request-payment', params: { listingId, bookingStart, bookingEnd, paymentMethod: stripePaymentMethodId }})

Reviewer: You could break pretty much all the examples into multiple lines for readability.

Author: I didn't notice that the extra "=>" string was a problem for Prettier.

```

> Note: card (StripeElement), and paymentParams are not needed when
> using previously saved payment card. FTW handles this in
> [`handleCardPayment` thunk function](https://github.com/sharetribe/flex-template-web/blob/master/src/ducks/stripe.duck.js#L657).

Stripe's frontent script checks if PaymentIntent needs extra actions
from customer. Some payments might need Strong Customer Authentication
(SCA). In practice, Stripe's script creates a popup (iframe) to card
issuers site, where 3D secure v2 authentication flow can be completed.

#### Step 3.

```js
=> sdk.transactions.transition({ id: transactionId, transition: 'transition/confirm-payment', params })
```

Inform Marketplace API, that PaymentIntent is ready to be captured after
possible SCA authentication has been requested from user. FTW does that
in
[`confirmPayment` thunk call](https://github.com/sharetribe/flex-template-web/blob/master/src/containers/CheckoutPage/CheckoutPage.duck.js#L206)

#### Step 4.

```js
=> sdk.messages.send({ transactionId: orderId, content: message })
```

FTW sends an initial message to transaction if customer has added a
message.

#### Step 5.

> As a final ste, we need to save the payment method, if customer has

Reviewer: ```suggestion
As a final step, we need to save the payment method, if customer has
```

selected the "Save for later use" checkbox. So, this is relevant if user
has selected onetime payment - instead of making a charge from the
previously saved credit card.

On FTW, we call

> [avePaymentMethod function](https://github.com/sharetribe/flex-template-web/blob/master/src/ducks/paymentMethods.duck.js#L200)

Reviewer: ```suggestion
[savePaymentMethod function](https://github.com/sharetribe/flex-template-web/blob/master/src/ducks/paymentMethods.duck.js#L200)
```

that creates stripe customer and adds updates default payment method.

There are 3 different scenarios, which require different calls to Flex
API:

**1. No StripeCustomer entity connected to Flex API:**

```js

> => sdk.stripeCustomer.create({ stripePaymentMethodId }, { expand: true, include: ['defaultPaymentMethod'] })

Reviewer: Is `{ stripePaymentMethodId }` the same as `{ stripePaymentMethodId: stripePaymentMethodId }`?

Author: Yes
